api/t8_mesh_handle/element.hxx

```diff
@@ -0,0 +1,260 @@
+   */
+  element (mesh<SelfType>* mesh, t8_locidx_t tree_id, t8_locidx_t element_id)
+    : m_mesh (mesh), m_tree_id (tree_id), m_element_id (element_id)
+  {
+  }
+
+  // --- Functions to check if caches exist. ---
+  /**
+   * Function that checks if a cache for the vertex coordinates exists.
+   * \return true if a cache for the vertex coordinates exists, false otherwise.
+   */
+  static constexpr bool
+  has_vertex_cache ()
+  {
+    return vertex_cache_exists;
+  }
+
+  /**
+   * Function that checks if a cache for the centroid exists.
+   * \return true if a cache for the centroid exists, false otherwise.
+   */
+  static constexpr bool
+  has_centroid_cache ()
+  {
+    return centroid_cache_exists;
+  }
+
+  // --- Functionality of the element. In each function, it is checked if a cached version exists (and is used then). ---
+  /**
+   * Getter for the refinement level of the mesh element.
+   * For this easily accessible variable, it makes no sense to provide a cached version.
+   * \return Refinement level of the mesh element.
+   */
+  t8_element_level
+  get_level () const
+  {
+    const t8_eclass_t tree_class = get_tree_class ();
+    const t8_element_t* element = get_element ();
+    return t8_forest_get_scheme (m_mesh->m_forest)->element_get_level (tree_class, element);
+  }
+
+  /**
+   * Getter for the vertex coordinates of the mesh element.
+   * This function uses or sets the cached version defined in TCompetence if available and calculates if not.
+   * \return Vector with one coordinate array for each vertex of the element.
+   */
+  std::vector<t8_3D_vec>
+  get_vertex_coordinates () const
+  {
+    // Check if we have a cached version and if the cache has already been filled.
+    if constexpr (vertex_cache_exists) {
+      if (this->vertex_cache_filled ()) {
+        return this->m_vertex_coordinates;
+      }
+    }
+    // Calculate the vertex coordinates.
+    const t8_element_t* element = get_element ();
+    const int num_corners
+      = t8_forest_get_scheme (m_mesh->m_forest)->element_get_num_corners (get_tree_class (), element);
+    std::vector<t8_3D_vec> vertex_coordinates;
+    vertex_coordinates.reserve (num_corners);
+    for (int icorner = 0; icorner < num_corners; ++icorner) {
+      t8_3D_vec vertex;
+      t8_forest_element_coordinate (m_mesh->m_forest, m_tree_id, element, icorner, vertex.data ());
+      vertex_coordinates.push_back (vertex);
+    }
+    // Fill the cache in the cached version.
+    if constexpr (vertex_cache_exists) {
+      this->m_vertex_coordinates = std::move (vertex_coordinates);
+      return this->m_vertex_coordinates;
+    }
+    return vertex_coordinates;
+  }
+
+  /**
+   * Getter for the center of mass of the mesh element.
+   * This function uses the cached version defined in TCompetence if available and calculates if not.
+   * \return Coordinates of the center.
+   */
+  t8_3D_vec
```

Comment:
```suggestion
  t8_3D_point
```

Reply:
I fixed this thing in #1938, but you are righty, it is better to fix it here. Just to let you know that i did not ignore your in person comment. :)